plasma_framework/test/src/vaults/EthVault.test.js

```diff
@@ -0,0 +1,123 @@
+const RLP = artifacts.require("RLP");
+const OutputModel = artifacts.require("OutputModel");
+const TransactionModel = artifacts.require("TransactionModel");
+const BlockController = artifacts.require("BlockController");
+const EthVault = artifacts.require("EthVault");
+
+const { BN, constants, expectRevert } = require('openzeppelin-test-helpers');
+const { expect } = require('chai');
+
+const Testlang = require("../helpers/testlang.js")
+const Transaction = require("../helpers/transaction.js").Transaction
+const TransactionOutput = require("../helpers/transaction.js").TransactionOutput
+
+contract("EthVault", accounts => {
+    const alice = accounts[1];
+    const DepositValue = 1000000;
+
+    before("setup libs", async () => {
+        const rlpLib = await RLP.new();
+        OutputModel.link("RLP", rlpLib.address);
+        const outputModel = await OutputModel.new();
+
+        TransactionModel.link("RLP", rlpLib.address);
+        TransactionModel.link("TransactionOutput", outputModel.address);
+        const transactionModel = await TransactionModel.new();
+        EthVault.link("TransactionModel", transactionModel.address);
+    });
+
+    beforeEach("setup contracts", async () => {
+        this.blockController = await BlockController.new(10);
+        this.ethVault = await EthVault.new(this.blockController.address);
+        await this.blockController.registerVault(1, this.ethVault.address);
+    });
+
+    describe("deposit", () => {
+
+        it("should store ethereum deposit", async () => {
+            let nextDepositBlock = parseInt(await this.blockController.nextDepositBlock(), 10);
+            expect(nextDepositBlock).to.be.equal(1);
+
+            const deposit = Testlang.deposit(DepositValue, alice);
+            await this.ethVault.deposit(deposit, {from: alice, value: DepositValue});
+            nextDepositBlock = parseInt(await this.blockController.nextDepositBlock(), 10);
+            expect(nextDepositBlock).to.be.equal(2);
+        });
+
+        it("should charge eth from depositing user", async () => {
+            const preDepositBalance = await web3.eth.getBalance(alice);
+            const deposit = Testlang.deposit(DepositValue, alice);
+            await this.ethVault.deposit(deposit, {from: alice, value: DepositValue});
+
+            const actualPostDepositBalance = new BN(await web3.eth.getBalance(alice));
+            const expectedPostDepositBalanceUpperBound =
```

Comment:
Another option is to calculate gas spent and take that into account when checking what the balance should be.
This is what I did in the `omg-js` tests, it's more exact than worrying about what gas _might_ cost.

It would look something like this:
```
            const { receipt } = await this.ethVault.deposit(deposit, {from: alice, value: DepositValue});

            const tx = await web3.eth.getTransaction(receipt.transactionHash)
            const spentOnGas = web3.utils.toBN(tx.gasPrice).muln(receipt.gasUsed)

            const actualPostDepositBalance = new BN(await web3.eth.getBalance(alice));
            
            const expectedPostDepositBalance =
                (new BN(preDepositBalance)).sub(new BN(DepositValue)).sub(spentOnGas);

            expect(actualPostDepositBalance).to.be.bignumber.equal(expectedPostDepositBalance);
```

Getting gas spent could be wrapped into a helper function e.g.
https://github.com/omisego/omg-js/blob/3218acfb200676b31e956d48b74b86f82505c9a6/packages/integration-tests/helpers/rootChainHelper.js#L52

Reply:
thanks, works perfect!